### PR TITLE
Improve multi-stage Python example for #309

### PR DESCRIPTION
**Pull request by lathama**
Question raised about the example which was incomplete. This commit improves the example some. Based off my example at https://lathama.net/git/lathama/Dockerfiles/src/branch/master/distroless/cherrypyexample/Dockerfile

### CODE REVIEW

File: examples/python3/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,14 @@
-FROM python:3-slim AS build-env
+FROM python:3.6-stretch AS build-env
```
Comment (briandealwis) [inline examples/python3/Dockerfile]: Why pin specifically 3.6?  And why `-slim` &rarr; `-stretch`?

File: examples/python3/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
 ADD . /app
 WORKDIR /app
+RUN pip3 install --upgrade pip
```
Comment (briandealwis) [inline examples/python3/Dockerfile]: Would it make sense to run `python -m compileall /app` too?

File: examples/python3/Dockerfile (hunk 1 above)
Comment (lathama) [reply examples/python3/Dockerfile]: I will have to check but distroless is still built off of stretch right. I was attempting to give some context here as it is only an example. Happy to make changes in the future.

File: examples/python3/Dockerfile [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
+RUN pip3 install -r requirements.txt
```
Comment (chanseokoh) [inline examples/python3/Dockerfile]: (I am not versed in Python.) This directory does not contain `requirements.txt`. Does this work OK without the file?

Comment (lathama) [reply examples/python3/Dockerfile]: Many assumptions here, yes. see the link in opening comment for context. Long term I suspect we need an actual example .py file for this demo along with tons of comments in the Dockerfile.

so adding everything from current project directory to the /app, change dir to the /app then use the projects requirements.txt

File: examples/python3/Dockerfile (hunk 2 above)
Comment (lathama) [reply examples/python3/Dockerfile]: That could be an option. I don't want to distract the user too much but I do see your point.

File: examples/python3/Dockerfile (hunk 3 above)
Comment (chanseokoh) [reply examples/python3/Dockerfile]: Sorry, I didn't get it. The example in the opening comment does have the `requirements.txt`: https://lathama.net/git/lathama/Dockerfiles/src/branch/master/distroless/cherrypyexample/requirements.txt

However, as you can see, our Distroless repo example doesn't have the file: https://github.com/GoogleContainerTools/distroless/tree/master/examples/python3

Can you confirm `docker built .` on this example won't fail on this `pip3 install -r requirements.txt` line?

**Comment by lathama** [comment]
My initial change was a quick copy and paste from https://lathama.net/git/lathama/Dockerfiles/src/branch/master/distroless/cherrypyexample to explain the usage a little. If you would like to have a more complete example I can do that but will need to sit down and work it out later this week.

**Comment by chanseokoh** [comment]
Sounds good. Just make sure anyone checking out the repo example and doing `docker build -t .` on this directory won't fail. And the example doesn't need to be complex. Try to keep it simple.

File: examples/python3/Dockerfile (hunk 3 above)
Comment (lathama) [reply examples/python3/Dockerfile]: I understand what you are asking. I will work up a more complete example later this week.

**Comment by lathama** [comment]
closing for more complete example later

**Pull request closed without merging**